miden-lib/asm/miden/contracts/wallets/basic.masm

```diff
@@ -73,3 +73,27 @@ export.send_asset.1
     # prepare the stack for return - stack has 6 elements too many
     movupw.3 dropw swap drop swap drop
 end
+
+#! Increments the account nonce by the provided value.
+#!
+#! Stack: [value]
+#! Output: []
+#!
+#! - value is the value to increment the nonce by. value can be at most 2^32 - 1 otherwise this
+#!   procedure panics.
+export.incr_nonce
+    exec.account::incr_nonce
+    # => []
+end
```

Comment:
Adding this procedure would lead to a security vulnerability: it will allow anyone to bypass an account's authentication mechanism. What is the motivation for adding it?

Reply:
The authentication mechanism is done in the kernel:

https://github.com/0xPolygonMiden/miden-base/blob/89437d17d4f490cc9b85562dae890eaa32c86dad/miden-lib/asm/kernels/transaction/api.masm#L131-L132

This is not skipping it (just like the other methods in this file)

Reply:
I mean the transaction authentication (e.g., where we require the user to provide a signature against an account's public key). Exposing this method publicly would allow anyone to bypass signature verification and update the account state at will.

Reply:
Ah, so the plan is to have the authentication code calling the nonce increment? That makes sense, I was unaware of that flow.

Reply:
Moved it back to the test only.